src/main/java/com/ghusse/ci/violations/bbcloud/lib/client/implementation/ClientV2.java

```diff
@@ -52,10 +52,20 @@ public void setAuthentication(String userName, String password) {
             description.getId(),
             page);
 
-    InputStream content = this.client.get(url);
+    InputStream content = null;
+    try {
+      content = this.client.get(url);
+    } catch (RestClientException e) {
+      throw new ClientException("Error while requesting the api.", description, e);
+    }
 
     TypeReference<PaginatedResponse<Comment>> type = new TypeReference<PaginatedResponse<Comment>>() {};
-    PaginatedResponse<Comment> pageComments = this.mapper.readValue(content, type);
+    PaginatedResponse<Comment> pageComments = null;
+    try {
+      pageComments = this.mapper.readValue(content, type);
+    } catch (IOException e) {
+      e.printStackTrace();
+    }
 
     if (pageComments.getPageLength() > page) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') NullPointerException might be thrown as 'pageComments' is nullable here [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS2259)